aiohttp/web_request.py

```diff
@@ -494,6 +497,47 @@ def if_unmodified_since(self) -> Optional[datetime.datetime]:
         """
         return self._http_date(self.headers.get(hdrs.IF_UNMODIFIED_SINCE))
 
+    @staticmethod
+    def _etag_values(etag_header: str) -> Iterator[ETag]:
+        """Extract `ETag` objects from raw header."""
+        if etag_header == ETAG_ANY:
+            yield ETag(
+                is_weak=False,
+                value=ETAG_ANY,
```

Comment:
There is some inconsistency here. If ETag is `"*"`, the value attribute will contain quotes. In all other cases they will be stripped.

Reply:
Not quite. `*` is a kind of special value, which is used **without** quotes. So if someone somehow will use `"*"` instead of `*`, it will be considered as regular etag.

Unfortunatly `ETag(value='*')` produced from `"*"` is indistinguishable from one produced from `*`.
So, for example:
```
"*", "tag", "other-tag" -> (ETag(value='*'), ETag(value='other-tag'))
* -> (ETag(value='*'), )
```
Maybe it's better to create separate type `ETAG_ANY` and change `Optional[Tuple[ETag, ...]]` to `Optional[Union[ETAG_ANY, Tuple[ETag, ...]]]`
```
"*", "tag", "other-tag" -> (ETag(value='*'), ETag(value='other-tag'))
* -> ETAG_ANY
```

@asvetlov what do you think?

Reply:
You are right.